x-pack/test/functional/apps/transform/actions/reauthorizing.ts

```diff
@@ -0,0 +1,216 @@
+        current_user: USER.TRANSFORM_VIEWER,
+      },
+    ];
+
+    before(async () => {
+      const apiKeyForTransformUsers =
+        await transform.securityCommon.createApiKeyForTransformUsers();
+
+      apiKeyForTransformUsers.forEach(({ user, apiKey }) =>
+        apiKeysForTransformUsers.set(user.name as USER, apiKey)
+      );
+
+      await esArchiver.loadIfNeeded('x-pack/test/functional/es_archives/ml/ecommerce');
+      await transform.testResources.createIndexPatternIfNeeded('ft_ecommerce', 'order_date');
+
+      for (const testData of testDataList) {
+        await transform.api.createTransform(testData.originalConfig.id, testData.originalConfig, {
+          deferValidation: true,
+          // Create transforms with secondary authorization headers
+          headers: generateHeaders(apiKeysForTransformUsers.get(testData.created_by_user)!),
+        });
+        // For transforms created with insufficient permissions, they can be created but not started
+        // so we should not assert that the api call is successful here
+        await transform.api.startTransform(testData.originalConfig.id, false);
+      }
+      await transform.testResources.setKibanaTimeZoneToUTC();
+    });
+
+    after(async () => {
+      await transform.securityCommon.clearAllTransformApiKeys();
+
+      for (const testData of testDataList) {
+        await transform.testResources.deleteIndexPatternByTitle(testData.originalConfig.dest.index);
+        await transform.api.deleteIndices(testData.originalConfig.dest.index);
+      }
+
+      await transform.api.cleanTransformIndices();
+      await transform.testResources.deleteIndexPatternByTitle('ft_ecommerce');
+    });
+
+    for (const testData of testDataList) {
+      const transformId = testData.originalConfig.id;
+
+      describe(`${testData.suiteTitle}`, function () {
+        it('reauthorize transform', async () => {
+          await transform.securityUI.loginAs(testData.current_user);
+
+          await transform.testExecution.logTestStep('should load the home page');
+          await transform.navigation.navigateTo();
+          await transform.management.assertTransformListPageExists();
+
+          await transform.testExecution.logTestStep('should display the transforms table');
+          await transform.management.assertTransformsTableExists();
+
+          await transform.testExecution.logTestStep(
+            'should display the transforms reauthorize callout'
+          );
+          await transform.management.assertTransformsReauthorizeCalloutExists();
+
+          await transform.testExecution.logTestStep(
+            'should display the original transform in the transform list'
+          );
+          await transform.table.filterWithSearchString(transformId, 1);
+          await transform.table.assertTransformRowFields(
+            transformId,
+            testData.expected.originalState
+          );
+
+          if (testData.expected.reauthorizeEnabled) {
```

Comment:
I think it would be worth to check that the row action is actually disabled for `reauthorizeEnabled: false`.

Reply:
Done here [`cc8cddc` (#156699)](https://github.com/elastic/kibana/pull/156699/commits/cc8cddca35ab6f69e5bba60bb534ba358063f644)